### PR TITLE
[로또] 최효원 미션 제출합니다.

### PR DESCRIPTION
**Pull request by chyo1**

### CODE REVIEW

File: src/main/java/lotto/Executor.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package lotto;
+
+import java.util.List;
+
+public class Executor {
+    private static final int LOTTO_PRICE = 1000;
+
+    IOController ioController;
+    Validator validator;
+    Parser parser;
```
Comment (Seol-JY) [inline src/main/java/lotto/Executor.java:10]: 접근 제어자가 생략되어 있네요! 캡슐화를 위해 private final로 선언하는 것이 좋을 것 같아요. 의존성 주입 받는 필드들은 불변성을 보장하는 것이 안전하다고 개인적으로 생각해요.

File: src/main/java/lotto/Executor.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        List<Integer> winningNumbers = getWinningNumbers();
+        int bonusNumber = getBonusNumber(winningNumbers);
+        WinningLotto winningLotto = WinningLotto.of(winningNumbers, bonusNumber);
+
+        WinningStatisticsDto winningStatisticsDto = issuedLottos.calculateWinningResults(winningLotto);
+        ioController.printWinningResults(winningStatisticsDto.getWinningStatistics());
+
+        double lottoYield = winningStatisticsDto.getLottoYield(purchaseAmount);
+        ioController.printWinningStatistics(lottoYield);
+    }
```
Comment (Seol-JY) [inline src/main/java/lotto/Executor.java:33]: run() 메서드가 한 가지 책임을 넘어서는 것 같아요. 메서드를 의미 있는 단위로 분리하면 가독성과 유지보수성이 더 좋아질 것 같습니다! 예를 들어 로또 발행, 당첨 번호 처리, 통계 계산 등으로 나눌 수 있겠네요.

File: src/main/java/lotto/IOController.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.List;
+import java.util.Map;
+
+public class IOController {
+    private static String INPUT_PURCHASE_AMOUNT_MESSAGE = "구입금액을 입력해 주세요.";
+    private static String INPUT_WINNING_NUMBERS_MESSAGE = "당첨 번호를 입력해 주세요.";
+    private static String INPUT_BONUS_NUMBER_MESSAGE = "보너스 번호를 입력해 주세요.";
+
+    private static String OUTPUT_PURCHASE_COUNT_MESSAGE = "개를 구매했습니다.";
+    private static String OUTPUT_WINNING_STATISTICS_MESSAGE = "당첨 통계\n---";
```
Comment (Seol-JY) [inline src/main/java/lotto/IOController.java:14]: 상수 필드에 final 키워드가 빠져있는 것 같아요! 불변성을 보장하기 위해 추가해 주면 좋을 것 같아요.

File: src/main/java/lotto/IOController.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public void printWinningResults(Map<Rank, Integer> winningStatistics) {
+        System.out.println(OUTPUT_WINNING_STATISTICS_MESSAGE);
+        winningStatistics.entrySet().stream()
+            .filter(entry -> entry.getKey() != Rank.ETC)
+            .sorted(Comparator.comparingInt(entry -> entry.getKey().ordinal()))
+            .forEach(entry -> {
+                System.out.println(entry.getKey().getPrintMessage() + " - " + entry.getValue() + "개");
+        });
```
Comment (Seol-JY) [inline src/main/java/lotto/IOController.java:67]: 해당 메서드에서 스트림 체이닝이 다소 복잡해 보이는 것 같아요! 가독성을 위해 중간 연산들을 별도 메서드로 추출하면 좋을 것 같습니다.

File: src/test/java/lotto/LottoTest.java [hunk 5]
```diff
@@ -21,5 +25,56 @@ class LottoTest {
...
+        int testBonusNumber = 7;
+
+        WinningLotto testWinningLotto = WinningLotto.of(testWinningNumber, testBonusNumber);
+
+        int expectedMatchCount = 6;
+    	//when
+        int actualMatchCount = testLotto.getMatchCount(testWinningLotto);
+
+        //then
+        assertEquals(expectedMatchCount, actualMatchCount);
```
Comment (Seol-JY) [inline src/test/java/lotto/LottoTest.java:78]: `보너스_번호_일치_여부_확인` 테스트에서 실제로 보너스 번호 일치 여부를 확인하지 않고 matchCount만 검증하고 있는 것 같아요! 의도한 테스트 케이스와 다른 것 같습니다!

File: src/test/java/lotto/WinningStatisticsDtoTest.java
Comment (Seol-JY) [inline src/test/java/lotto/WinningStatisticsDtoTest.java:1]: boundary case에 대한 테스트가 더 추가되면 좋을 것 같습니다!

- 구매 금액이 0인 경우
- 최대 당첨금이 발생하는 경우
- 소수점 자리수 처리가 정확한지 등을 검증하면 좋을 것 같아요.

**Review by Seol-JY — COMMENTED** [review]
3주차도 정말 고생 많으셨습니다~! 
마지막 주차도 화이팅입니다!

File: src/main/java/lotto/Executor.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        return parser.parseBonusNumber(bonusNumberInput);
+    }
+
+    private List<Integer> getWinningNumbers() {
+        String winningNumbersInput;
+        while (true) {
+            try {
+                winningNumbersInput = ioController.inputWinningNumbers();
+                validator.validateWinningNumbers(winningNumbersInput);
+                break;
```
Comment (speculatingwook) [inline src/main/java/lotto/Executor.java:69]: 여기서 break를 걸지 않고 바로 값을 return해도 정상적으로 작동합니다!

File: src/main/java/lotto/Executor.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        WinningLotto winningLotto = WinningLotto.of(winningNumbers, bonusNumber);
+
+        WinningStatisticsDto winningStatisticsDto = issuedLottos.calculateWinningResults(winningLotto);
+        ioController.printWinningResults(winningStatisticsDto.getWinningStatistics());
+
+        double lottoYield = winningStatisticsDto.getLottoYield(purchaseAmount);
+        ioController.printWinningStatistics(lottoYield);
+    }
+
+    private int getPurchaseAmount() {
```
Comment (speculatingwook) [inline src/main/java/lotto/Executor.java:35]: 이 로직들은 IOController가 있는데 , 여기에 있는 것 보다 IOController쪽에 두는 게 더 낫지 않나 생각합니다!

File: README.md
Comment (speculatingwook) [inline README.md:1]: 객체지향적으로 작성하려고 노력하신 부분들이 많이 보입니다! 제안드리고 싶은 건 각 역할에 맞게 디렉토리로 나누는 것입니다! 제 레포에 들어와 보시면 역할에 맞게 디렉토리 분리를 해두었는데, 디렉토리로 한번 나누기만 해도 가독성이 크게 올라갑니다!

**Review by speculatingwook — COMMENTED** [review]
객체지향의 관점에서는 나름대로 잘 작성하셨다고 생각합니다! 이번주도 화이팅입니다!
